### ISSUE TITLE
You are not allowed to transition layouts when the image is not acquired.

### ISSUE THREAD

**Issue by ratchetfreak** [SaschaWillems/Vulkan#156]
https://www.khronos.org/registry/vulkan/specs/1.0-wsi_extensions/xhtml/vkspec.html#_wsi_swapchain

> Use of a presentable image must occur only after the image is returned by vkAcquireNextImageKHR, and before it is presented by vkQueuePresentKHR. **This includes transitioning the image layout and rendering commands.**

https://github.com/SaschaWillems/Vulkan/blob/master/base/vulkanswapchain.hpp#L374

This will end up doing an image transition on a swapchain image that is not owned by the application. Instead you can always transition from UNDEFINED in the pre-present transition.

**Comment by SaschaWillems** [comment]
Thx for pointing me at that part of the spec. When I wrote the swapchain class the only example code available was the one from the LunarG examples and they also do this.

But I wonder why validation doesn't catch this (and why it works on all GPUs I have tested so far).

If such an image layout transition is not allowed for images not owned by the application (which makes sense) then there should be a validation that catches this.
